quickwit/quickwit-proto/src/lib.rs

```diff
@@ -297,11 +296,60 @@ pub fn set_parent_span_from_request_metadata(request_metadata: &tonic::metadata:
     Span::current().set_parent(parent_cx);
 }
 
+/// Index identifiers that uniquely identify not only the index, but also
+/// its incarnation allowing to distinguish between deleted and recreated indexes.
+/// It is represented as a stiring in index_id:incarnation_id format.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct IndexUid(String);
```

Comment:
I didn't, but I don't see how this will be better than what I had here - https://github.com/quickwit-oss/quickwit/blob/main/quickwit/quickwit-metastore/src/metastore/index_metadata/mod.rs#L40